code/modules/mob/living/brain/posibrain.dm

```diff
@@ -41,15 +41,26 @@ GLOBAL_VAR(posibrain_notify_cooldown)
 			GLOB.posibrain_notify_cooldown = world.time + askDelay
 
 /obj/item/device/mmi/posibrain/attack_self(mob/user)
-	if(!brainmob || brainmob.key)
+	if(!brainmob)
+		to_chat(user, "<span class='warning'>[src]'s personality matrix is corrupted! Attempting to reboot...</span>")
+		brainmob = new(src)
+		if(!brainmob) //something went really wrong!
```

Comment:
The check was initially in the `attack_self` but had no message, so I assumed that it was possible somehow and should be accounted for. I'll remove it, but I'm still cautious about it.